### ISSUE TITLE
[feature request] Even more fine-grained theorem callout referencing settings

### ISSUE THREAD

**Issue by joelatschool** [RyotaUshio/obsidian-math#208]
Sorry to be asking for something so specific but I don't see how to achieve it without a change to Math Booster and I think it makes a lot of sense:

I'd like for it to be possible to make it so that a a theorem without a title is linked by `[note title] > [type] [number]` but a theorem with a title is linked by `[title]`.  This might be best achieved with a template system, if that's possible.

**Comment by RyotaUshio** [comment]
Thanks again for the helpful suggestion.

For now, I'm thinking of something like this.
- In the plugin settings, users can configure four different templates for theorem referencing format:
    1. For a link of the form `[[note#^block-id]]` pointing to a theorem with its own title
    2. For a link of the form `[[note#^block-id]]` pointing to a theorem without title
    3. For a link of the form `[[#^block-id]]` pointing to a theorem with its own title
    4. For a link of the form `[[#^block-id]]` pointing to a theorem without title
- In templates, users can use variables like
    - `{{note}}`: note title (only applicable for 1 & 2)
    - `{{type}}`: theorem callout type, e.g. Theorem, Definition, etc
    - `{{type-number}}`: `{{type}}` with the theorem number, e.g. Theorem 1.1, Definition A, etc. For unnumbered theorems, this is exactly the same as `{{type}}`.
    - `{{title}}`: theorem title, e.g. "Cauchy-Schwarz inequation" for "Theorem 1.1 (Cauchy-Schwarz inequation)."

What do you think?

**Comment by lwzbr** [comment]
Yes, I like this idea. I would like to be able to suppress the note title automatically if the theorem callout has a title, but not under other circumstances.

For example, I might have a note called “Pythagorean Theorem” that contains a theorem callout called “Pythagorean Theorem” as well as some commentary, examples, etc. In this case linking to [[Pythagorean Theorem#^theorem-id]] will produce “Pythagorean Theorem > Pythagorean Theorem”, which is redundant. I would like to suppress the note title in this instance.

But, in other circumstances, I might want to reference an unnamed lemma or proposition from a different note. In this case, it would be helpful to have “Note Title > Proposition 4” or something like that.

Right now “show note title at the head of a link” is a blanket toggle, but its usefulness is situation-dependent.

**Comment by JaimeVicent** [comment]
@lwzbr maybe you find out this snippet helpful [link](https://publish.obsidian.md/slrvb-docs/ITS+Theme/Embed+Adjustments). With it you can embed a note and add an attribute to it so the filename doesn't appear. Use this syntax: `![[note|no-title]]`
